src/EditorFeatures/Core/EncapsulateField/AbstractEncapsulateFieldCommandHandler.cs

```diff
@@ -121,14 +121,15 @@ public bool ExecuteCommand(EncapsulateFieldCommandArgs args, CommandExecutionCon
             if (previewService != null)
             {
                 await _threadingContext.JoinableTaskFactory.SwitchToMainThreadAsync(cancellationToken);
-                finalSolution = previewService.PreviewChanges(
+                finalSolution = await previewService.PreviewChangesAsync(
                     string.Format(EditorFeaturesResources.Preview_Changes_0, EditorFeaturesResources.Encapsulate_Field),
                      "vs.csharp.refactoring.preview",
                     EditorFeaturesResources.Encapsulate_Field_colon,
                     result.Name,
                     result.Glyph,
                     finalSolution,
-                    solution);
+                    solution,
+                    cancellationToken).ConfigureAwait(false);
```

Comment:
Looks like this needs to be ConfigureAwait(true) or else the undo manager stuff below will run off the UI thread?

Reply:
(blocking for this reason)

Reply:
Good catch 🖤